src/apps/apploader/app.js

```diff
@@ -80,58 +80,79 @@ define(function(require) {
 		 */
 		render: function(pArgs) {
 			var self = this,
-				args = pArgs || {},
-				template;
-
-			if (!self.isRendered()) {
-				self.getUserApps(function(appList) {
-					if (monster.config.whitelabel.useDropdownApploader) {
-						template = $(self.getTemplate({
-							name: 'appList',
-							data: {
-								defaultApp: appList[0],
-								apps: appList,
-								allowAppstore: monster.apps.auth.currentUser.priv_level === 'admin' && !monster.config.whitelabel.hideAppStore
-							}
-						}));
+				globalCallback = _.get(pArgs, 'callback');
 
-						$('#appList').empty().append(template);
+			monster.waterfall([
+				function maybeFetchApps(callback) {
+					if (self.isRendered()) {
+						return callback(true);
+					}
+					self.getUserApps(_.partial(callback, null));
+				},
+				function insertAppLinks(appList, callback) {
+					callback(null, self.insertAppLinks(appList));
+				},
+				function renderTemplate(appList, callback) {
+					var templateTypes = {
+							dropdown: {
+								name: 'appList',
+								defaultApp: _.head(appList),
+								insert: function(template) {
+									self.bindDropdownApploaderEvents(template);
+
+									$('#appList').empty().append(template);
+								}
+							},
+							modal: {
+								name: 'app',
+								defaultApp: monster.ui.formatIconApp(appList[0]),
+								insert: function(template) {
+									self.bindEvents(template, appList);
 
-						self.bindDropdownApploaderEvents(template);
-					} else {
-						template = $(self.getTemplate({
-							name: 'app',
-							data: {
+									self.appFlags.modal = monster.ui.fullScreenModal(template, {
+										hideClose: true,
+										destroyOnClose: false
+									});
+								}
+							}
+						},
+						templateData = _.get(templateTypes, monster.config.whitelabel.useDropdownApploader ? 'dropdown' : 'modal'),
+						$template = $(self.getTemplate({
+							name: templateData.name,
+							data: _.merge({
 								allowAppstore: monster.apps.auth.currentUser.priv_level === 'admin' && !monster.config.whitelabel.hideAppStore,
-								defaultApp: monster.ui.formatIconApp(appList[0]),
 								apps: appList
-							}
+							}, _.pick(templateData, [
+								'defaultApp'
+							]))
 						}));
 
-						self.bindEvents(template, appList);
+					templateData.insert($template);
 
-						self.appFlags.modal = monster.ui.fullScreenModal(template, {
-							hideClose: true,
-							destroyOnClose: false
-						});
-					}
-					args.callback && args.callback();
-				});
-			} else {
-				self.show({
-					callback: args.callback
-				});
-			}
+					callback(null, appList);
```

Comment:
`appList` is not being used in the final callback for `monster#waterfall`. I think it can be safely removed from these callback arguments.